components/Tip.tsx

```diff
@@ -0,0 +1,84 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import {
+  Box,
+  Typography,
+  Button,
+  Paper,
+  Stack,
+  IconButton,
+  useTheme,
+} from '@mui/material';
+import { Lightbulb, ChevronDown } from 'react-bootstrap-icons';
+
+interface TipProps {
+  title: string;
+  text: string;
+  storageKey: string;
+}
+
+export const Tip = ({ title, text, storageKey }: TipProps) => {
+  const theme = useTheme();
+  const [isTextVisible, setIsTextVisible] = useState(true);
+  const [isTipOpen, setIsTipOpen] = useState(true);
+
+  useEffect(() => {
+    const savedTipState = localStorage.getItem(storageKey);
```

Comment:
We should wrap `localStorage` functions in a `try ... catch` block, local storage may be undefined and throw an error. Here's an example of that in the [`RefreshPrompt` component in Paths](https://github.com/kausaltech/kausal-paths-ui/blob/main/src/components/general/RefreshPrompt.tsx#L12), keeping the local storage calls in separate functions also makes it easier to test (in future)